arelle/ModelRelationshipSet.py

```diff
@@ -98,6 +98,12 @@ def baseSetRelationship(arcElement):
             return rel
     return None
 
+def baseSetSortkey(baseSetKey): # for base set sorting functions
+    arcrole, linkrole, _linkqname, _arcqname = baseSetKey
+    # arcrole may be a string or tuple of strings.
+    return (",".join(arcrole) if isinstance(arcrole, (tuple,list)) else arcrole or "",
+            linkrole or "")
```

Comment:
Would it be more robust to normalize to `list`/`tuple` rather than joining with an arbitrary delimiter?

Reply:
we need it lexicographic, where some newer base sets have multiple alternative arcroles (calc, TLB, probably others).  It was doing as array/string before but no longer works where calc 1.1 has multiple in some base sets and single in other.

Reply:
I believe `list<list` is still lexicographic for inner strings (`['a']<['b']` vs `['b']<['a']`), it just generally seems cleaner (avoids needing to pick an unused delimiter; `['a','b']<['a,b']` produces different result than joining).

Reply:
This is going out at the SEC as is. But they should address this in their next update